src/commands.ts

```diff
@@ -1150,31 +1154,30 @@ export const insertTableFootnote = (
   const insertedAt = state.selection.to
 
   const inlineFootnotes = findChildrenByType(node, schema.nodes.inline_footnote)
-  const labels = inlineFootnotes.map(
-    (nodeWithPos) => nodeWithPos.node.attrs.contents
-  )
-  const inlineFootnoteNode = state.schema.nodes.inline_footnote.create({
-    rids: [footnote.attrs.id],
-    contents: labels.length ? Math.max(...labels) + 1 : 1, // I need to revisit this
-  }) as InlineFootnoteNode
+  const footnoteIndex =
+    inlineFootnotes.filter(({ pos }) => position + pos <= insertedAt).length + 1
 
   const tr = state.tr
 
-  // insert the inline footnote
-  tr.insert(insertedAt, inlineFootnoteNode)
+  let insertionPos = position
 
-  let insertionPos
   const footnotesElement = findChildrenByType(
     node,
     schema.nodes.footnotes_element
-  ).pop()
+  )[0]
```

Comment:
This meant to be 'pop()' to retrieve the last item on the list. This approach is necessary to handle scenarios where there are multiple footnote elements, but some have been rejected and are not currently visible. By retrieving the last item, we can accurately represent the latest change made, ensuring correct handling of rejected elements.

Reply:
reverted